### PR TITLE
Tweak the module scapetable declaration to match the cpp header decls

### PR DESCRIPTION
**Pull request by ghost**
It looks like the header declaration cant be changed to match the D version so tweak the D version instead.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/dmodule.d [hunk 1]
```diff
@@ -429,6 +429,8 @@ extern (C++) final class Module : Package
         if (doHdrGen)
             hdrfile = setOutfile(global.params.hdrname, global.params.hdrdir, arg, global.hdr_ext);
         //objfile = new File(objfilename);
+        escapetable = new Escape();
+        memset(escapetable, 0, Escape.sizeof);
```
Comment (Basile-z) [inline src/dmd/dmodule.d]: This memset is not useful i believe

File: src/dmd/dmodule.d [hunk 2]
```diff
@@ -400,7 +400,7 @@ extern (C++) final class Module : Package
     Identifiers* versionidsNot; // forward referenced version identifiers
 
     Macro* macrotable;          // document comment macros
-    Escape escapetable;        // document comment escapes
+    Escape* escapetable;        // document comment escapes
```
Comment (ZombineDev) [inline src/dmd/dmodule.d]: IIRC, the C++ version used `Escape **`, can you update it to `Escape*` as well?

**Review by PetarKirov — APPROVED** [review]

File: src/dmd/dmodule.d (hunk 2 above)
Comment (Basile-z) [reply src/dmd/dmodule.d]: No, it was already good, see https://github.com/dlang/dmd/pull/9418/files/50d758aa1ce8f7f5d073a357da5acbcc88171b40#diff-5c1e8d0f4cb0f0151c3b5d3dc6dce89f

File: src/dmd/dmodule.d [hunk 3]
```diff
@@ -429,6 +429,7 @@ extern (C++) final class Module : Package
         if (doHdrGen)
             hdrfile = setOutfile(global.params.hdrname, global.params.hdrdir, arg, global.hdr_ext);
         //objfile = new File(objfilename);
+        escapetable = new Escape();
```
Comment (jacob-carlborg) [inline src/dmd/dmodule.d]: What's the advantage of using a pointer which requires dynamically allocate an object?

Comment (Basile-z) [reply src/dmd/dmodule.d]: It's not about pro and cons here, it's about the cpp version that does not allow a module to store a Escape as value so it's made like that.

**Comment by Basile-z** [comment]
ping for merge, this is a fixup so either this has to be merged or the other PR reverted. Thanks !

**Review by jacob-carlborg — APPROVED** [review]

**Comment by thewilsonator** [comment]
Done.

**Pull request merged**